### PR TITLE
RIP-230 Copying /public/canvas/ from junction to ripley and updating …

### PR DESCRIPTION
**Pull request by mlincoln**
…accessibility footer link

https://jira-secure.berkeley.edu/browse/RIP-230

### CODE REVIEW

File: public/canvas/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  var loadCanvasCustomization = function() {
+    window.CALCENTRAL = 'https://junction.berkeley.edu';
+
+    // Ensure the bCourses development and test servers are pointing to the correct
+    // CalCentral instance when a copy of production is made
+    if (window.location.origin === 'https://ucberkeley.beta.instructure.com') {
+      window.CALCENTRAL = 'https://junction-dev.berkeley.edu';
+    } else if (window.location.origin === 'https://ucberkeley.test.instructure.com') {
+      window.CALCENTRAL = 'https://junction-qa.berkeley.edu';
```
Comment (johncrossman) [inline public/canvas/index.js]: In the URLs above, replace `junction` with `ripley`. Secondly, replace `berkeley.edu` with `ets.berkeley.edu`.

File: public/canvas/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+(function(window, $) {
+  'use strict';
+
+  var loadCanvasCustomization = function() {
+    window.CALCENTRAL = 'https://junction.berkeley.edu';
```
Comment (johncrossman) [inline public/canvas/index.js]: @mlincoln - Ideally, we rename `window.CALCENTRAL` to `window.RIPLEY`.

File: public/canvas/canvas-customization.js [hunk 3]
```diff
@@ -0,0 +1,864 @@
...
+  /**
+   * Get the id for a custom LTI tool that user has access to
+   *
+   * @param  {String}     toolType            The classification of the LTI tool. This will be the top level property in the external tools hash
+   * @param  {String}     toolName            The name of the LTI tool for which the id should be retrieved
+   * @param  {Function}   callback            Standard callback function
+   * @param  {String}     callback.id         The id of the requested custom LTI tool
+   */
+  var getExternalToolId = function(toolType, toolName, callback) {
+    apiRequest('/api/academics/canvas/external_tools.json', function(externalToolsHash) {
```
Comment (johncrossman) [inline public/canvas/canvas-customization.js]: @mlincoln - this is a Ripley API reference and it should be changed to `/api/canvas/external_tools`

File: public/canvas/canvas-customization.js [hunk 4]
```diff
@@ -0,0 +1,864 @@
...
+  /* CREATE A SITE */
+
+  /**
+   * Check whether the current user is allowed to create a new site
+   *
+   * @param  {Function}   callback                  Standard callback function
+   * @param  {Element}    callback.canCreateSite    Whether the current user is allowed to create a new site
+   */
+  var canUserCreateSite = function(callback) {
+    apiRequest('/api/academics/canvas/user_can_create_site?canvas_user_id=' + window.ENV.current_user_id, function(authResult) {
```
Comment (johncrossman) [inline public/canvas/canvas-customization.js]: @mlincoln - This API path has changed to `/api/canvas/can_user_create_site`

File: public/canvas/canvas-customization.js
Comment (mlincoln) [inline public/canvas/canvas-customization.js:1]: @johncrossman should this one be changed to /api/canvas/egrade_export/is_official_course.json ?

File: public/canvas/canvas-customization.js [hunk 5]
```diff
@@ -0,0 +1,864 @@
...
+
+  /**
+   * Add the 'E-Grades' export option to the Canvas Gradebook
+   */
+  var addEGrades = function() {
+    // Verify that the current context is the Gradebook tool
+    if (window.ENV && window.ENV.GRADEBOOK_OPTIONS && window.ENV.GRADEBOOK_OPTIONS.context_id) {
+      // Verify that the current course contains official course sections
+      var courseId = window.ENV.GRADEBOOK_OPTIONS.context_id;
+      apiRequest('/api/academics/canvas/egrade_export/is_official_course.json?canvas_course_id=' + courseId, function(officialCourseResponse) {
```
Comment (johncrossman) [inline public/canvas/canvas-customization.js:295]: @mlincoln - I'm not sure we have this API endpoint written yet. Check `canvas_site_controller.py`

File: public/canvas/canvas-customization.js [hunk 6]
```diff
@@ -0,0 +1,864 @@
...
+   * Customize the default footer with Berkeley information
+   */
+  var customizeFooter = function() {
+    // Replace the Instructure logo with the Berkeley logo
+    var $berkeleyLogo = $('<a>', {
+      'class': 'footer-logo',
+      'href': 'http://www.berkeley.edu',
+      'title': 'University of California, Berkeley',
+      'css': {
+        'backgroundImage': 'url(' + window.CALCENTRAL + '/static/canvas/images/ucberkeley_footer.png)'
```
Comment (johncrossman) [inline public/canvas/canvas-customization.js]: @mlincoln - This URL reference to this PNG (image) will need to be altered.

File: public/canvas/index.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    // Load the JavaScript customizations
+    $.getScript(window.CALCENTRAL + '/static/canvas/canvas-customization.js');
+
+    // Load the CSS customizations
+    var css = $('<link>', {
+      'rel': 'stylesheet',
+      'type': 'text/css',
+      'href': window.CALCENTRAL + '/static/canvas/canvas-customization.css'
```
Comment (johncrossman) [inline public/canvas/index.js]: @mlincoln - I think my PR review is done. Keep an eye out for any use of `window.CALCENTRAL` (as seen above).  Those URLs will likely need to be modified to the Ripley way.

Comment (mlincoln) [reply public/canvas/index.js]: I missed this one. Will amend again. 

Waiting to see if my .gitignore change addressed the travis failures.

File: .gitignore [hunk 8]
```diff
@@ -38,3 +38,7 @@ tmp/
 .cache
 .flask_cache/
 .pytest_cache/
+
+# JS beasts
+public/canvas/canvas-customization.js
+public/canvas/index.js
```
Comment (johncrossman) [inline .gitignore]: @mlincoln - You do NOT want to use `.gitignore` for the purpose of do-not-lint. I would remove the lines above. Instead, modify the `.eslintrc.js` file with guidance from https://eslint.org/docs/latest/use/configure/ignore

Comment (mlincoln) [reply .gitignore]: Got it. If my change to .eslintrc.js satisfies travis, I'll squash the double commit.

**Pull request merged**